presto-hive/src/test/java/com/facebook/presto/hive/TestHiveUtil.java

```diff
@@ -116,6 +144,38 @@ private static void assertToPartitionValues(String partitionName)
         assertEquals(actual, expected);
     }
 
+    private static void assertToPartitionNamesAndValues(String partitionName, List<String> expectedKeyList)
+            throws MetaException
+    {
+        Map<String,String> actual = toPartitionNamesAndValues(partitionName);
+        AbstractList<String> expectedValueList = new ArrayList<>();
+        for( String s: expectedKeyList) {
+            expectedValueList.add(null);
+        }
+        Warehouse.makeValsFromName(partitionName, expectedValueList);
+        boolean equalMaps = true;
+
+        if(actual.keySet().size() != expectedKeyList.size()){
+            equalMaps = false;
+        } else {
+            for (int index = 0; index < expectedKeyList.size(); index++) {
+                String key = expectedKeyList.get(index);
+                if (!actual.containsKey(key)) {
+                    equalMaps = false;
+                    break;
+                }
+                String actualValue = actual.get(key);
+                String expectedValue = expectedValueList.get(index);
+                if (!actualValue.equals(expectedValue)) {
+                    equalMaps = false;
+                    break;
+                }
+            }
+        }
+
+        assertTrue(equalMaps);
```

Comment:
rather checking in the end, may we should assert it when the equal fails at first spot?